.github/workflows/compile-requirements.yml

```diff
@@ -79,4 +79,4 @@ jobs:
           find temp-requirements -name "requirements-*" -type f -exec mv -f -t requirements {} +
           git add -A
           git commit -m "Automatically generated requirements"
-          git push -f https://x-access-token:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
+          until git push -f https://x-access-token:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git; do git pull --rebase; done
```

Comment:
Could we just `git pull --rebase` once and then `git push` once, or force this job to run after the compile jobs complete? I don't fully understand why we need to pull/rebase in a loop here. Is there a possibility of this running indefinitely if the pull/rebase fails for some reason?

Reply:
There shouldn't by any possibility of running indefinitely. The push in job B will fail if job A pushes after job B rebases but before job B pushes, and there is a certain probability of this happening that is non-zero. Because of the way the jobs are self-contained (and running on different platforms), we cannot combine the commits and then push them one by one (or altogether), this is why every job has it's own "commit and push" step.

The loop is to handle the job A and job B example case I illustrated above, and in theory it could run up to two times (at least in the current setup) in one job (e.g., if two different jobs rebase and push between the rebase and the push of the third job).

Reply:
In theory, we could create temporary branches where we push these commits to, and then combine them all in a different job, but I don't think it's necessarily worth that complexity at this point in time.